File: examples/basic-client.ts
import { Wazeko } from "../packages/wazeko/src/index.js";

async function main() {
  console.log("Starting Wazeko Client (TypeScript)...");

  const client = Wazeko.builder()
    .authStore("./auth")
    .printQr(true)
    .build();

  await client.connect();

  for await (const event of client.events()) {
    console.log(`[Event Received] ${event.name}:`, event.data);
  }
}

main().catch(console.error);

File: crates/wazeko/examples/basic_client.rs
use wazeko::Wazeko;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    tracing_subscriber::fmt::init();

    println!("Starting Wazeko Client...");
    let client = Wazeko::builder()
        .auth_store("./auth")
        .print_qr(true)
        .build();

    client.connect().await?;

    let mut events = client.events();

    while let Some(event) = events.recv().await {
        println!("[Event Received] {event:?}");
    }

    Ok(())
}
